data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorkerTest.java

```diff
@@ -0,0 +1,169 @@
+                                            final S3SelectResponseHandler selectResponseHandlerTest,
+                                            final S3ObjectPluginMetrics s3ObjectPluginMetrics,
+                                            final CompressionType compressionType) {
+        if(selectResponseHandlerTest == null)
+            selectObjectContentResponse(responseFormat);
+        Random random = new Random();
+        numberOfRecordsToAccumulate = random.nextInt(10) + 2;
+        bucketName = UUID.randomUUID().toString();
+        key = UUID.randomUUID().toString();
+        when(s3ObjectReference.getBucketName()).thenReturn(bucketName);
+        when(s3ObjectReference.getKey()).thenReturn(key);
+        when(s3SourceConfig.getS3SelectOptions()).thenReturn(s3SelectOptions);
+        when(s3SourceConfig.getS3SelectOptions().getQueryStatement()).thenReturn(queryStatement);
+        S3ObjectRequest request = new S3ObjectRequest.Builder(buffer,numberOfRecordsToAccumulate,
+                bufferTimeout,s3ObjectPluginMetrics)
+                .queryStatement(queryStatement).eventConsumer(eventConsumer)
+                .serializationFormatOption(format).fileHeaderInfo(FileHeaderInfo.NONE)
+                .s3AsyncClient(s3AsyncClient).compressionType(compressionType)
+                .s3SelectResponseHandler(selectResponseHandler).build();
+        return new S3SelectObjectWorker(request);
+    }
+    @Test
+    void selectObjectFromS3CsvTestWithEmptyResponse() {
+        when(s3ObjectPluginMetrics.getS3ObjectsFailedCounter()).thenReturn(s3ObjectsFailedCounter);
+        S3SelectResponseHandler selectResponseHandler =new S3SelectResponseHandler();
+        final S3SelectObjectWorker selectObjectUnderTest = createSelectObjectUnderTest(null, "select * from s3Object", S3SelectSerializationFormatOption.CSV,
+                selectResponseHandler, s3ObjectPluginMetrics, CompressionType.NONE);
+        final ArgumentCaptor<SelectObjectContentRequest> request = ArgumentCaptor
+                .forClass(SelectObjectContentRequest.class);
+        final IOException exception = assertThrows(IOException.class, () -> selectObjectUnderTest.parseS3Object(s3ObjectReference));
+        assertNotNull(exception);
+        verify(s3AsyncClient).selectObjectContent(request.capture(), any(S3SelectResponseHandler.class));
+        assertThat(request.getValue().key(), equalTo(key));
+        assertThat(request.getValue().bucket(), equalTo(bucketName));
+        assertThat(request.getValue().expectedBucketOwner(), nullValue());
+        assertThat(request.getValue().expression(), equalTo(s3SourceConfig.getS3SelectOptions().getQueryStatement()));
+        verify(s3ObjectsFailedCounter).increment();
+        assertTrue(selectResponseHandler.getReceivedEvents().isEmpty());
+    }
+    @ParameterizedTest
+    @CsvSource({
+            "{\"name\":\"data-prep\"},select * from s3Object,CSV,NONE",
```

Comment:
It would be good to include some tests with multiple objects in the response as well.

Reply:
Thanks David for your review, Changes are incorporated and available for review.